pkg/specgen/generate/kube/kube.go

```diff
@@ -971,10 +980,10 @@ func getPodPorts(containers []v1.Container) []types.PortMapping {
 			}
 			// only hostPort is utilized in podman context, all container ports
 			// are accessible inside the shared network namespace
-			if p.HostPort != 0 {
+			if p.HostPort != 0 && p.ContainerPort != 0 {
```

Comment:
Looking at the code of the loop, this can only happen if both `ContainerPort` and `HostPort` were not set in the YAML file. I think it will be cleaner to check it in the beginning of the loop and `continue` instead of going through all the steps just to ignore them

Reply:
Good catch, thanks!